agreementbot/api.go

```diff
@@ -442,6 +446,31 @@ func (a *API) agreement(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// Obscure the secret details for all secrets in the given proposal
+func obscureSecretDetails(p *persistence.Agreement) error {
+	prop := p.Proposal
+
+	var tsandcs map[string]interface{}
+	if err := json.Unmarshal([]byte(prop), &tsandcs); err != nil {
+		return err
+	} else if err := json.Unmarshal([]byte(tsandcs["tsandcs"].(string)), &tsandcs); err != nil {
+		return err
+	} else if _, ok := tsandcs["secretDetails"]; !ok {
+		// there are no secrets to obscure, so return nil
+		return nil
+	}
+	secretDetails := tsandcs["secretDetails"].([]interface{})
+
+	// loop over secrets, if any
+	for _, secretDetail := range secretDetails {
```

Comment:
As part of my PR for issue 2475 I had to rework the data structure for secretDetails. You will have to change this code to adapt to the new changes. My previous comment should make this easier.